src/components/Signup/Signup.js

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { FormattedHTMLMessage, injectIntl } from 'react-intl';
+import { timeout } from '../../utils';
+import { Formik, Form } from 'formik';
+import { FieldGroup, InputFieldItem, CheckboxFieldItem } from '../form-helpers/form-helpers';
+
+const fieldNames = {
+  firstname: 'firstname',
+  lastname: 'lastname',
+  phone: 'phone',
+  email: 'email',
+  password: 'password',
+  accept_privacy_policies: 'accept_privacy_policies',
+  accept_promotions: 'accept_promotions',
+};
+
+/** Prepare empty values for all fields
+ * It is required because in another way, the fields are not marked as touched.
+ */
+const getFormInitialValues = () =>
+  Object.keys(fieldNames).reduce(
+    (accumulator, currentValue) => ({ ...accumulator, [currentValue]: '' }),
+    {},
+  );
+
+export default injectIntl(function({ intl }) {
+  const _ = (id, values) => intl.formatMessage({ id: id }, values);
+
+  const validate = (values) => {
+    const errors = {};
+    if (!values[fieldNames.firstname]) {
+      errors[fieldNames.firstname] = _('validation_messages.error_required_field');
+    }
+
+    if (!values[fieldNames.lastname]) {
+      errors[fieldNames.lastname] = _('validation_messages.error_required_field');
+    }
+
+    if (!values[fieldNames.phone]) {
+      errors[fieldNames.phone] = _('validation_messages.error_required_field');
+    } else {
+      // TODO: validate phone
+    }
+
+    if (!values[fieldNames.email]) {
+      errors[fieldNames.email] = _('validation_messages.error_required_field');
+    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values[fieldNames.email])) {
+      // TODO: review if this is our desired validation
```

Comment:
use the same as doppler not?

Reply:
Good catch! No, I do not think so. It should. 

I will try to fix in this PR or at least adding a TODO.

Reply:
Oh, sorry, the TODO is already there.